### PR TITLE
updated SignalFX migration guide

### PR DESCRIPTION
**Pull request by kubawach**
SignalFX -> OTEL migration guide updated with latest config properties and some minor spelling / formatting fixes.

### CODE REVIEW

File: MIGRATING.MD [hunk 1]
```diff
@@ -3,20 +3,22 @@
...
 [SignalFx Java Agent](https://github.com/signalfx/signalfx-java-tracing).
 If you're using the SignalFx Java Agent, migrate to the Splunk Distribution of
-OpenTelemetry Java Instrumentation to use OpenTelemetry's instrumentation to
-send traces to Splunk APM. The Splunk Distribution of OpenTelemetry Java
-Instrumentation uses OpenTelemetry to instrument applications, which is an
-open-source API to gather telemetry data, and has a smaller footprint.
+OpenTelemetry Java Instrumentation.
+
+The distribution uses [OpenTelemetry Java Instrumentation](https://github.com/open-telemetry/opentelemetry-java-instrumentation) 
+to instrument applications, which in open-source, robust and has a smaller memory footprint.
```
Comment (mateuszrzeszutek) [inline MIGRATING.MD:9]: Nitpicking a bit, but how about

```suggestion
The Splunk distribution is based on the [OpenTelemetry Java Instrumentation](https://github.com/open-telemetry/opentelemetry-java-instrumentation):
an open-source, robust project that uses OpenTelemetry API and has a smaller memory
footprint than the SignalFx Java Agent.
```

File: MIGRATING.MD [hunk 2]
```diff
@@ -3,20 +3,22 @@
...
+
+## Tracing data
```
Comment (mateuszrzeszutek) [inline MIGRATING.MD:11]: Hmm, how about `Telemetry changes`?

File: MIGRATING.MD [hunk 3]
```diff
@@ -25,22 +27,35 @@ against the data you used to receive from the SignalFx Java Agent.
...
+1. Download the [latest release](https://github.com/signalfx/splunk-otel-java/releases/latest/download/splunk-otel-javaagent-all.jar)
+   of the Splunk Distribution of OpenTelemetry Java Instrumentation. For example use: 
+   ```bash
+   $ # download the newest version of the agent
+   $ sudo curl -vsSL -o splunk-otel-javaagent-all.jar 'https://github.com/signalfx/splunk-otel-java/releases/latest/download/splunk-otel-javaagent-all.jar'
    ```
+2. Set the service name. This is how you can identify the service in Splunk APM.
+
+   Example how you can set it with an environment variable with a
+   service name of `yourServiceName` with environment variable:
```
Comment (mateuszrzeszutek) [inline MIGRATING.MD:39]:
```suggestion
   For example, you can set the service name using an environment variable:
```

File: MIGRATING.MD [hunk 4]
```diff
@@ -25,22 +27,35 @@ against the data you used to receive from the SignalFx Java Agent.
...
+   ```bash
    $ EXPORT OTEL_RESOURCE_ATTRIBUTES=service.name=my-java-app
    ```
+   or with system property:
```
Comment (mateuszrzeszutek) [inline MIGRATING.MD:43]:
```suggestion
   or a system property:
```

File: MIGRATING.MD [hunk 5]
```diff
@@ -25,22 +27,35 @@ against the data you used to receive from the SignalFx Java Agent.
...
+   ```
+   -Dotel.resource.attributes=service.name=my-java-app
+   ```
 3. Specify the endpoint of the SignalFx Smart Agent or OpenTelemetry Collector
    you're exporting traces to. You can set the endpoint with a system property
-   or environment variable. This is how you can set it with an environment
-   variable with an endpoint of `http://yourEndpoint:9080/v1/trace`:
+   or environment variable.
```
Comment (mateuszrzeszutek) [inline MIGRATING.MD:49]:
```suggestion
   or an environment variable. 
```

File: MIGRATING.MD [hunk 6]
```diff
@@ -25,22 +27,35 @@ against the data you used to receive from the SignalFx Java Agent.
...
+   ```
+   -Dotel.resource.attributes=service.name=my-java-app
+   ```
 3. Specify the endpoint of the SignalFx Smart Agent or OpenTelemetry Collector
    you're exporting traces to. You can set the endpoint with a system property
-   or environment variable. This is how you can set it with an environment
-   variable with an endpoint of `http://yourEndpoint:9080/v1/trace`:
+   or environment variable. 
+   
+   Example how you can set it with an environment variable with an endpoint of `http://yourEndpoint:9080/v1/trace`:
```
Comment (mateuszrzeszutek) [inline MIGRATING.MD:51]:
```suggestion
   Assuming your endpoint is `http://yourEndpoint:9080/v1/trace`, this is how you set it with an environment variable:
```

File: MIGRATING.MD [hunk 7]
```diff
@@ -25,22 +27,35 @@ against the data you used to receive from the SignalFx Java Agent.
...
    ```
    $ EXPORT OTEL_EXPORTER_JAEGER_ENDPOINT="http://yourEndpoint:9080/v1/trace"
    ```
+   or with system property:
```
Comment (mateuszrzeszutek) [inline MIGRATING.MD:55]:
```suggestion
   or a system property:
```

File: MIGRATING.MD [hunk 8]
```diff
@@ -64,32 +79,37 @@ OpenTelemetry Java Instrumentation.
 ### Configuration setting changes
 
 These SignalFx Java Agent system properties correspond to the following
-OpenTelemetry system properties:
+OpenTelemetry system properties (NOTE: for exporter - specific properties we use `jaeger` as an example):
```
Comment (mateuszrzeszutek) [inline MIGRATING.MD:82]: Should we even mention other exporters when jaeger is the default?

File: MIGRATING.MD [hunk 9]
```diff
@@ -64,32 +79,37 @@ OpenTelemetry Java Instrumentation.
...
 
 | SignalFx system property | OpenTelemetry system property |
 | ------------------------ | ----------------------------- |
-| `signalfx.service.name` | `otel.resource.attributes=service.name=my-java-app` |
+| `signalfx.service.name` | `otel.resource.attributes=service.name=<name of the service>` |
+| `signalfx.env` | `otel.resource.attributes=deployment.environment=<name of the environment>` |
```
Comment (mateuszrzeszutek) [inline MIGRATING.MD:87]: Please add a footnote explaining that when setting both you need to merge them into a single env var/property

File: MIGRATING.MD [hunk 10]
```diff
@@ -64,32 +79,37 @@ OpenTelemetry Java Instrumentation.
...
 | `signalfx.endpoint.url` | `otel.exporter.jaeger.endpoint` |
-| `signalfx.tracing.enabled` | `otel.trace.enabled` |
-| `signalfx.span.tags` | `otel.resource.attributes` |
-| `signalfx.recorded.value.max.length` | `otel.config.max.attr.length` |
-| `signalfx.db.statement.max.length` | `otel.config.max.attr.length` | 
+| `signalfx.tracing.enabled` | `otel.javaagent.enabled` |
+| `signalfx.integration.<name>.enabled=false` | `otel.instrumentation.[id].enabled=false` |
```
Comment (mateuszrzeszutek) [inline MIGRATING.MD:90]: Should we keep the template <name> the same on both sides?

```suggestion
| `signalfx.integration.<name>.enabled=false` | `otel.instrumentation.<name>.enabled=false` | 
```

File: MIGRATING.MD [hunk 11]
```diff
@@ -64,32 +79,37 @@ OpenTelemetry Java Instrumentation.
...
+| `signalfx.tracing.enabled` | `otel.javaagent.enabled` |
+| `signalfx.integration.<name>.enabled=false` | `otel.instrumentation.[id].enabled=false` | 
+| `signalfx.span.tags` | `otel.resource.attributes=<comma separated key=value pairs>` |
 | `signalfx.trace.annotated.method.blacklist` | `otel.trace.annotated.methods.exclude` |
 | `signalfx.trace.methods` | `otel.trace.methods` |
-| `signalfx.integration.<name>.enabled=true` | `otel.integration.[id].enabled=false` | 
+| `signalfx.server.timing.context` | `splunk.context.server-timing.enabled` |
+
+
+Additional info about disabling a particular instrumentations can be found in the [OpenTelemetry Java Instrumentation docs](https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/main/docs/suppressing-instrumentation.md)
```
Comment (mateuszrzeszutek) [inline MIGRATING.MD:97]:
```suggestion
Additional info about disabling a particular instrumentations can be found in the [OpenTelemetry Java Instrumentation docs](https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/main/docs/suppressing-instrumentation.md).
```

File: MIGRATING.MD [hunk 12]
```diff
@@ -64,32 +79,37 @@ OpenTelemetry Java Instrumentation.
...
 
 These SignalFx Java Agent environment variables correspond to the following
 OpenTelemetry environment variables:
 
 | SignalFx environment variable | OpenTelemetry environment variable |
 | ----------------------------- | ---------------------------------- |
 | `SIGNALFX_SERVICE_NAME` | `OTEL_RESOURCE_ATTRIBUTES=service.name=my-java-app` |
```
Comment (mateuszrzeszutek) [inline MIGRATING.MD:104]:
```suggestion
| `SIGNALFX_SERVICE_NAME` | `OTEL_RESOURCE_ATTRIBUTES=service.name=<name of the service>` |
```

File: MIGRATING.MD [hunk 13]
```diff
@@ -64,32 +79,37 @@ OpenTelemetry Java Instrumentation.
...
+| `SIGNALFX_ENV` | `OTEL_RESOURCE_ATTRIBUTES=deployment.environment=<name of the environment>` |
 | `SIGNALFX_ENDPOINT_URL` |`OTEL_EXPORTER_JAEGER_ENDPOINT` |
-| `SIGNALFX_TRACING_ENABLED` | `OTEL_TRACE_ENABLED` |
+| `SIGNALFX_TRACING_ENABLED` | `OTEL_JAVAAGENT_ENABLED` |
+| `SIGNALFX_INTEGRATION_<name>_ENABLED=false` | `OTEL_INSTRUMENTATION_[id]_ENABLED=false` |
```
Comment (mateuszrzeszutek) [inline MIGRATING.MD:108]:
```suggestion
| `SIGNALFX_INTEGRATION_<name>_ENABLED=false` | `OTEL_INSTRUMENTATION_<name>_ENABLED=false` |
```

File: MIGRATING.MD [hunk 14]
```diff
@@ -64,32 +79,37 @@ OpenTelemetry Java Instrumentation.
...
 | `SIGNALFX_SPAN_TAGS` | `OTEL_RESOURCE_ATTRIBUTES` |
-| `SIGNALFX_RECORDED_VALUE_MAX_LENGTH` | `OTEL_CONFIG_MAX_ATTR_LENGTH` |
 | `SIGNALFX_TRACE_ANNOTATED_METHOD_BLACKLIST` | `OTEL_TRACE_ANNOTATED_METHODS_EXCLUDE` |
 | `SIGNALFX_TRACE_METHODS` | `OTEL_TRACE_METHODS` |
+| `SIGNALFX_SERVER_TIMING_CONTEXT` | `SPLUNK_CONTEXT_SERVER-TIMING_ENABLED` |
```
Comment (mateuszrzeszutek) [inline MIGRATING.MD:112]:
```suggestion
| `SIGNALFX_SERVER_TIMING_CONTEXT` | `SPLUNK_CONTEXT_SERVER_TIMING_ENABLED` |
```

**Comment by mateuszrzeszutek** [comment]
Also please rebase - there's been a couple of changes to *.md files in the last few days.

File: MIGRATING.MD (hunk 1 above)
Comment (kubawach) [reply MIGRATING.MD:9]: Quite good actually 👍

File: MIGRATING.MD (hunk 2 above)
Comment (kubawach) [reply MIGRATING.MD:11]: The idea is to have a separate section describing the data (perhaps extended with a table comparing main differences). But will change to "Telemetry data" as it sounds better.

File: MIGRATING.MD (hunk 3 above)
Comment (kubawach) [reply MIGRATING.MD:39]: It goes like "an example how to do it this... or this... way". So I prefer "This is an example" rather "for example". But will add an article as it is missing now (in both cases).

File: MIGRATING.MD (hunk 5 above)
Comment (kubawach) [reply MIGRATING.MD:49]: Done.

File: MIGRATING.MD (hunk 10 above)
Comment (kubawach) [reply MIGRATING.MD:90]: In OTEL instrumentation it's called "instrumentation ID" so let's stay coherent here.

File: MIGRATING.MD (hunk 11 above)
Comment (kubawach) [reply MIGRATING.MD:97]: Heh xD

File: MIGRATING.MD (hunk 6 above)
Comment (kubawach) [reply MIGRATING.MD:51]: Done as with the previous one.

File: MIGRATING.MD (hunk 7 above)
Comment (kubawach) [reply MIGRATING.MD:55]: +1

File: MIGRATING.MD (hunk 8 above)
Comment (kubawach) [reply MIGRATING.MD:82]: I think so - if the customer selects other exporter suggested config change `SIGNALFX_ENDPOINT_URL` ->|`OTEL_EXPORTER_JAEGER_ENDPOINT` will not work.

File: MIGRATING.MD (hunk 9 above)
Comment (kubawach) [reply MIGRATING.MD:87]: Makes sense.

File: MIGRATING.MD (hunk 12 above)
Comment (kubawach) [reply MIGRATING.MD:104]: +1

File: MIGRATING.MD (hunk 13 above)
Comment (kubawach) [reply MIGRATING.MD:108]: Due to https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/main/docs/suppressing-instrumentation.md I'll stay with ID.

File: MIGRATING.MD (hunk 14 above)
Comment (kubawach) [reply MIGRATING.MD:112]: +1

**Comment by kubawach** [comment]
@mateuszrzeszutek @jkwatson pls have another look

File: smoke-tests/src/test/java/com/splunk/opentelemetry/GlassFishSmokeTest.java [hunk 15]
```diff
@@ -34,8 +34,7 @@
   private static Stream<Arguments> supportedConfigurations() {
     return configurations("payara")
         .otelLinux("5.2020.6", PAYARA_SERVER_ATTRIBUTES, VMS_ALL, "8", "11")
-        .otelWindows("5.2020.6", PAYARA_SERVER_ATTRIBUTES, VMS_ALL, "8", "11")
-        .stream();
+        .otelWindows("5.2020.6", PAYARA_SERVER_ATTRIBUTES, VMS_ALL, "8", "11").stream();
```
Comment (jkwatson) [inline smoke-tests/src/test/java/com/splunk/opentelemetry/GlassFishSmokeTest.java:37]: why did these smoke test changes make it into this PR?

File: MIGRATING.md [hunk 16]
```diff
@@ -64,32 +79,39 @@ OpenTelemetry Java Instrumentation.
...
-| `signalfx.service.name` | `otel.resource.attributes=service.name=my-java-app` |
+| `signalfx.service.name` | `otel.resource.attributes=service.name=<name of the service>` |
+| `signalfx.env` | `otel.resource.attributes=deployment.environment=<name of the environment>` |
 | `signalfx.endpoint.url` | `otel.exporter.jaeger.endpoint` |
-| `signalfx.tracing.enabled` | `otel.trace.enabled` |
-| `signalfx.span.tags` | `otel.resource.attributes` |
-| `signalfx.recorded.value.max.length` | `otel.config.max.attr.length` |
-| `signalfx.db.statement.max.length` | `otel.config.max.attr.length` | 
+| `signalfx.tracing.enabled` | `otel.javaagent.enabled` |
+| `signalfx.integration.<name>.enabled=false` | `otel.instrumentation.[id].enabled=false` |
```
Comment (jkwatson) [inline MIGRATING.md:90]: why the different formatting for placeholders here?

File: smoke-tests/src/test/java/com/splunk/opentelemetry/GlassFishSmokeTest.java (hunk 15 above)
Comment (kubawach) [reply smoke-tests/src/test/java/com/splunk/opentelemetry/GlassFishSmokeTest.java:37]: NO IDEA. Must have been a rebase thing, but I fail to understand how (did rebase on main only).

File: MIGRATING.md (hunk 16 above)
Comment (kubawach) [reply MIGRATING.md:90]: Copied from dependencies (signalfx / otel). Can make it uniform tho.

File: MIGRATING.md [hunk 17]
```diff
@@ -25,22 +28,34 @@ against the data you used to receive from the SignalFx Java Agent.
...
-   of the Splunk Distribution of OpenTelemetry Java Instrumentation.
-2. Set the service name. This is how you can identify the service in APM.
-   You can set the service name with a system property or environment
-   variable. This is how you can set it with an environment variable with a
-   service name of `yourServiceName`:
+1. Download the [latest release](https://github.com/signalfx/splunk-otel-java/releases/latest/download/splunk-otel-javaagent-all.jar)
+   of the Splunk Distribution of OpenTelemetry Java Instrumentation. For example use: 
+   ```bash
+   $ # download the newest version of the agent
+   $ sudo curl -vsSL -o splunk-otel-javaagent-all.jar 'https://github.com/signalfx/splunk-otel-java/releases/latest/download/splunk-otel-javaagent-all.jar'
```
Comment (mateuszrzeszutek) [inline MIGRATING.md:35]: You don't need `sudo` here

```suggestion
   $ curl -vsSL -o splunk-otel-javaagent-all.jar 'https://github.com/signalfx/splunk-otel-java/releases/latest/download/splunk-otel-javaagent-all.jar'
```

File: MIGRATING.md [hunk 18]
```diff
@@ -25,22 +28,34 @@ against the data you used to receive from the SignalFx Java Agent.
...
    ```
+2. Set the service name. This is how you can identify the service in Splunk APM.
+
+   An example how to set it using an environment variable:
+   ```bash
    $ EXPORT OTEL_RESOURCE_ATTRIBUTES=service.name=my-java-app
```
Comment (mateuszrzeszutek) [inline MIGRATING.md:41]:
```suggestion
   $ export OTEL_RESOURCE_ATTRIBUTES=service.name=my-java-app
```

File: MIGRATING.md [hunk 19]
```diff
@@ -25,22 +28,34 @@ against the data you used to receive from the SignalFx Java Agent.
...
+   ```
 3. Specify the endpoint of the SignalFx Smart Agent or OpenTelemetry Collector
    you're exporting traces to. You can set the endpoint with a system property
-   or environment variable. This is how you can set it with an environment
-   variable with an endpoint of `http://yourEndpoint:9080/v1/trace`:
+   or environment variable. 
+   
+   An example how to set it using an environment variable:
    ```
    $ EXPORT OTEL_EXPORTER_JAEGER_ENDPOINT="http://yourEndpoint:9080/v1/trace"
```
Comment (mateuszrzeszutek) [inline MIGRATING.md:53]:
```suggestion
   $ export OTEL_EXPORTER_JAEGER_ENDPOINT="http://yourEndpoint:9080/v1/trace"
```

File: MIGRATING.md [hunk 20]
```diff
@@ -64,32 +79,39 @@ OpenTelemetry Java Instrumentation.
 ### Configuration setting changes
 
 These SignalFx Java Agent system properties correspond to the following
-OpenTelemetry system properties:
+OpenTelemetry system properties (NOTE: some properites are exporter - specific, the default is `jaeger`):
```
Comment (mateuszrzeszutek) [inline MIGRATING.md:82]: Typo; remove spaces around the hyphen:

```suggestion
OpenTelemetry system properties (NOTE: some properties are exporter-specific, the default is `jaeger`):
```

> the default is `jaeger`

Default exporter? We use `jaeger-thrift-splunk` by default

File: MIGRATING.md [hunk 21]
```diff
@@ -64,32 +79,39 @@ OpenTelemetry Java Instrumentation.
...
+| `signalfx.span.tags` | `otel.resource.attributes=<comma separated key=value pairs>` |
 | `signalfx.trace.annotated.method.blacklist` | `otel.trace.annotated.methods.exclude` |
 | `signalfx.trace.methods` | `otel.trace.methods` |
-| `signalfx.integration.<name>.enabled=true` | `otel.integration.[id].enabled=false` | 
+| `signalfx.server.timing.context` | `splunk.context.server-timing.enabled` |
+
+Note: when setting both `service name` and `environment` appropriate `otel.resource.attributes` property setting will 
+look like this: `otel.resource.attributes=service.name=myService,deployment.environment=myEnvironment` 
+
+Additional info about disabling a particular instrumentations can be found in the [OpenTelemetry Java Instrumentation docs](https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/main/docs/suppressing-instrumentation.md).
```
Comment (mateuszrzeszutek) [inline MIGRATING.md:99]:
```suggestion
Additional info about disabling a particular instrumentation can be found in the [OpenTelemetry Java Instrumentation docs](https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/main/docs/suppressing-instrumentation.md).
```

File: MIGRATING.md (hunk 20 above)
Comment (kubawach) [reply MIGRATING.md:82]: Yes we do, but properties are generic for all jaeger-based exporters and stating full name here would confuse customers migrating their code.

File: MIGRATING.md (hunk 21 above)
Comment (kubawach) [reply MIGRATING.md:99]: +1

**Review by jkwatson — APPROVED** [review]

**Pull request merged**